hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockNodeConnectionManager.java

```diff
@@ -34,169 +46,162 @@
-            });
-
-            // Create BlockItemSet by adding all items at once
-            BlockItemSet itemSet =
-                    BlockItemSet.newBuilder().addAllBlockItems(protocBlockItems).build();
-
-            batchRequests.add(
-                    PublishStreamRequest.newBuilder().setBlockItems(itemSet).build());
-        }
-
-        // Stream prepared batches to each connection
-        for (BlockNodeConnection connection : connectionsToStream) {
-            final var connectionNodeConfig = connection.getNodeConfig();
-            try {
-                for (PublishStreamRequest request : batchRequests) {
-                    connection.sendRequest(request);
-                }
-                logger.info(
-                        "Sent block {} to stream observer for Block Node {}:{}",
-                        blockNumber,
-                        connectionNodeConfig.address(),
-                        connectionNodeConfig.port());
-            } catch (Exception e) {
-                logger.error(
-                        "Failed to send block {} to stream observer for Block Node {}:{}",
-                        blockNumber,
-                        connectionNodeConfig.address(),
-                        connectionNodeConfig.port(),
-                        e);
-            }
-        }
+    public void handleEndOfStreamSuccess(
+            @NonNull final BlockNodeConnection connection, @Nullable final Long blockNumber) {
+        requireNonNull(connection);
+        updateLatestBlock(connection, blockNumber);
     }
 
     /**
      * Initiates the streaming of a block to all active connections.
      *
      * @param block the block to be streamed
      */
-    public void startStreamingBlock(@NonNull BlockState block) {
-        streamingExecutor.execute(() -> streamBlockToConnections(block));
+    public void startStreamingBlock(@NonNull final BlockState block) {
+        streamingExecutor.execute(() -> streamBlock(block));
     }
 
     /**
      * Handles connection errors from a BlockNodeConnection by removing the failed connection
-     * and initiating the reconnection process.
      *
-     * @param node the node configuration for the failed connection
+     * @param connection the connection that received the error
+     * @param thrown the error that occurred
      */
-    public void handleConnectionError(@NonNull BlockNodeConfig node) {
+    public void handleConnectionError(
+            @NonNull final BlockNodeConnection connection,
+            // This isn't used yet, but will be once the code for responding to specific error codes is introduced
+            @SuppressWarnings("unused") @NonNull final Throwable thrown) {
         synchronized (connectionLock) {
-            activeConnections.remove(node); // Remove the failed connection
+            // If available, make the secondary connection the primary connection
+            if (connection == primary && !isRetrying(secondary) && !secondaryActive()) {
```

Comment:
Below `Objects.equals()` is used to compare connections and here `==` is used, is that on purpose?

Reply:
Nope, good catch 👍

Reply:
Fixed